source/common/network/connection_balancer_impl.cc

```diff
@@ -0,0 +1,40 @@
+#include "common/network/connection_balancer_impl.h"
+
+namespace Envoy {
+namespace Network {
+
+void ExactConnectionBalancerImpl::registerHandler(BalancedConnectionHandler& handler) {
+  absl::MutexLock lock(&lock_);
+  handlers_.push_back(&handler);
+}
+
+void ExactConnectionBalancerImpl::unregisterHandler(BalancedConnectionHandler& handler) {
+  absl::MutexLock lock(&lock_);
+  handlers_.erase(std::find(handlers_.begin(), handlers_.end(), &handler));
```

Comment:
should we keep the handlers in a set to avoid a potential scalability problem here?

alternatively you could keep the vector index in the object and erase by moving the last item, to avoid indexes getting stale.

Reply:
The number of listeners is typically low and this is a very rare operation, so I'm not sure it matters, but I'm happy to do this if you think it's best. I can also just add a comment about the tradeoffs. WDYT?

Reply:
commenting about tradeoffs is fine.